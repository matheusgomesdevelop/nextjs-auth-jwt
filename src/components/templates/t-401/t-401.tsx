// Atoms
import { A_Button, A_Text, A_Title } from "@/components/atoms";

// StyleSheets
import styles from "./t-401.module.css";

export interface I_T401 {
  handleLogin: () => void;
}

export const T_401 = ({ handleLogin }: I_T401) => {
  return (
    <main data-testid="t-401" className={styles.t_401} id="t-401">
      <A_Text variant="fwSb-fs16-primary">401</A_Text>
      <A_Title variant="h1">Acesso não autorizado</A_Title>
      <A_Text variant="fwReg-fs16-gray500">
        Desculpe, você não tem permissão para acessar esta página. Por favor, verifique suas credenciais ou entre em
        contato com o suporte se precisar de assistência.
      </A_Text>
      <A_Button variant="gradient" onClick={handleLogin}>
        Faça Login
      </A_Button>
    </main>
  );
};
