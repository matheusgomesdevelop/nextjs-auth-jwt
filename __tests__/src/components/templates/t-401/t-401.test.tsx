import renderer from 'react-test-renderer';

// Template
import { T_401 } from '@/components/templates/t-401/t-401';

describe('Deve renderizar o template 401 corretamente', () => {
    it('Deve preservar a estrutura do template 401', () => {
        const three = renderer.create(<T_401 handleLogin={() => null} />).toJSON();

        expect(three).toMatchSnapshot();
    });
});
