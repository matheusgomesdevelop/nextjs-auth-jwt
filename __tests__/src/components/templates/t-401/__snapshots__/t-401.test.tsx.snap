// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o template 401 corretamente Deve preservar a estrutura do template 401 1`] = `
<main
  data-testid="t-401"
>
  <p
    className="a_text__fwSb_fs16_primary"
    data-testid="a-text"
  >
    401
  </p>
  <h1
    className="a_title__h1"
    data-testid="a-title"
  >
    Acesso não autorizado
  </h1>
  <p
    className="a_text__fwReg_fs16_gray500"
    data-testid="a-text"
  >
    Desculpe, você não tem permissão para acessar esta página. Por favor, verifique suas credenciais ou entre em contato com o suporte se precisar de assistência.
  </p>
  <button
    className="a_button__gradient p-button p-component"
    data-pc-name="button"
    data-pc-section="root"
    data-testid="a-button"
    disabled={false}
    onClick={[Function]}
  >
    Faça Login
  </button>
</main>
`;
